### ISSUE TITLE
Story: Add External Link Icon to NCIDS Guide Cards

### ISSUE THREAD

**Issue by andyvanavery31** [NCIOCPL/ncids#1935]
### Specifications

**Visual**
- External Link icon is center aligned vertically on the link text and fixed on the right within the button itself
- If the link text would wrap, the last word of the title will wrap so that the icon stays fixed on the right where it is center aligned vertically

**Functional**
- If the link text does not have .gov it will get an external link icon
- A Guide Card button link can be overridden to have an external link icon
- A Guide Card button link with an external link icon can be overridden to remove it

**Impacted Components**

- [ ] Guide Card
  - [ ] Doc Site update for [Guide Card page](https://designsystem-dev.cancer.gov/develop/components/guide-card)
  - [ ] Storybook for [Default Guide Card](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-guide-card-default--default)
  - [ ] Storybook for [Guide Card with Image and Description](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-guide-card-variants-guide-card-with-image-and-description--single-guide-card-with-image-and-description-without-heading)

### DoR Tasks

- [x] Comps Created
- [x] Comps Reviewed using [Checklist](https://github.com/NCIOCPL/cgov-digital-platform/wiki/Product-Review-of-Design-Solutions)
  - [x] Developer Review
  - [x] IA Review
  - [x] Product Review
- [x] Comps Approved
  - [x] Developer Approval
  - [x] IA Approval
  - [x] Product Approval
- [x] Scenarios Created
- [x] Scenarios Reviewed 
  - [x] Designer Review
  - [x] Developer Review
  - [x] IA Review
- [x] Scenarios Approved
  - [x] Designer Approval
  - [x] Developer Approval
  - [x] IA Approval
- [x] Acceptance Criteria Drafted
- [x] Acceptance Criteria Reviewed
  - [x] Developer Review
  - [x] IA Review
- [x] Acceptance Criteria Approved
  - [x] Developer Approval
  - [x] IA Approval

### DoD Tasks

- [ ] Governance Drafted
- [ ] PR Review (see PR for review and comments)
- [ ] Client review with demo

### Technical Notes

**Scenarios/Comps**

- Use the comps here: [Guide Card with Image and Description with External Link Icon](https://www.figma.com/design/GCOniv8OTyj9OC1udKwo6S/Mini-Landing-Pages?node-id=9051-4829&node-type=text&m=dev). **NOTE** There is no comp for the Default Guide Card, however, the behavior for the external link icon on the Guide Card buttons is consistent.
- On the [NCIDS Documentation Site Guide Card page](https://designsystem-dev.cancer.gov/develop/components/guide-card) add:
  - In the [Best Practices section](https://designsystem-dev.cancer.gov/develop/components/guide-card#best-practices), add a subsection which will be the last subsection with the heading, "External Links", and in the subsequent paragraph explain how and when to use and add the External Link Icon. In this section, explain how the external links are mostly "auto-magical", and provide an explanation of how to use the override.
  - Two more [Code Snippet](https://designsystem-dev.cancer.gov/develop/components/guide-card#code-snippets) examples where one is a Default Guide Card and the other is a Guide Card with Image and Description. For each Guide Card:
      - Make the first link button as an external link that links to [www.google.com](http://www.google.com/), and has the button text "Link to Google".  
      - The second link button will be an external link that links to www.google.com, has the button text "External Link Icon Overridden", and the override class will be applied so that the external link icon does not appear.
      - The third link button will be an internal link that links to www.cancer.gov, has the button text "Internal Link with Icon Override", and the override class will be applied so that the external link icon appears.
- On Storybook:
  - Within the Guide Card folder, add two stories to the [Test Cases folder](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-guide-card-test-cases--guide-with-image-and-description-without-image). 
    - The first story will be "Default with External Link" 
      - It will have the first link button as an external link that links to [www.google.com](http://www.google.com/), and has the button text "Link to Google".  
      - The second link button will be an external link that links to www.google.com, has the button text "External Link Icon Overridden", and the override class will be applied so that the external link icon does not appear.
      - The third link button will be an internal link that links to www.cancer.gov, has the button text "Internal Link with Icon Override", and the override class will be applied so that the external link icon appears.
    - The second story will be "Guide Card with Image and Description with External link", and it will match the single guide card with image and description from the Doc Site.

### Acceptance Criteria

```gherkin
Scenario: An NCIDS Guide Card with an external link is viewed on the NCIDS
  Given the external link icon has been added to the NCIDS Documentation Site and NCIDS Storybook for the Guide Card
    And the Guide Card link button is linked to a url that does not contain `.gov`
  When the Guide Card link button is a linked url that does not contain `.gov`
  Then the Guide Card link button will appear with the external link icon
    And the icon on the Guide Card link button will match the comp (see links above)
    And the external link icon will remain center aligned vertically on the button that it is on
  When the text of the Guide Card link button wraps
  Then the external link icon will stay on the right side of the link button
  When a user wants to override the behavior of a Guide Card link button with a link that does not contain `.gov` to not display the external link icon
  Then they will override the behavior with an override class
    And the external link icon will not be displayed
  When a user has a Guide Card link button with a link that contains `.gov`
    And they want to override the behavior to include the external link icon
  Then they will override the behavior with an override class
    And the external link icon will be displayed
```

**Comment by bennettcc** [comment]
Per standup 1/3: this is awaiting merge of PR: https://github.com/NCIOCPL/ncids/pull/1951
